inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -19,15 +27,16 @@ def application_add(args: argparse.Namespace) -> str:
     manifest = ('<?xml version="1.0" encoding="utf-8"?>' +
                 '<manifest>' +
                 '<type>source</type>' +
-                '<applicationSource>' +
-                '<add><gpg>'
-                '{0}' +
-                '{1}'
-                '</gpg><repo><repos>').format(create_xml_tag(arguments, "uri"),
-                                              create_xml_tag(arguments, "keyname"))
+                '<applicationSource>' + '<add>')
 
-    for source in args.sources:
-        manifest += '<source_pkg>' + source.strip() + '</source_pkg>'
+    if args.gpgKeyUri and args.gpgKeyName:
+        manifest += ('<gpg>' + '{0}' + '{1}' + '</gpg>').format(create_xml_tag(arguments, "uri"),
+                                                                create_xml_tag(arguments, "keyname"))
```

Comment:
**Maintainability**
**Suggestion:** Avoid using string concatenation for XML elements to prevent potential formatting issues.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    gpg_element = f'<gpg>{create_xml_tag(arguments, "uri")}{create_xml_tag(arguments, "keyname")}</gpg>'
    manifest += gpg_element
```